be/src/exprs/vectorized/json_functions.cpp

```diff
@@ -137,23 +145,32 @@ Status JsonFunctions::extract_from_object(simdjson::ondemand::object& obj, const
         }
 
         if (index != -1) {
-            auto arr = tvalue.get_array();
-            if (arr.error()) {
+            // try to access tvalue as array.
+            simdjson::ondemand::array arr;
+            auto err = tvalue.get_array().get(arr);
+            if (err) {
                 return Status::InvalidArgument(fmt::format("failed to access field as array, field: {}, err: {}", col,
-                                                           simdjson::error_message(arr.error())));
+                                                           simdjson::error_message(err)));
             }
 
-            int idx = 0;
-            for (auto a : arr) {
-                if (a.error()) {
-                    return Status::InvalidArgument(fmt::format("failed to access array element, field: {}, err: {}",
-                                                               col, simdjson::error_message(arr.error())));
-                }
+            size_t sz;
+            err = arr.count_elements().get(sz);
+            if (err) {
+                return Status::InvalidArgument(
+                        fmt::format("failed to get array size, field: {}, err: {}", col, simdjson::error_message(err)));
+            }
 
-                if (idx++ == index) {
-                    a.get(tvalue);
-                    break;
-                }
+            if (index >= sz) {
+                return Status::InvalidArgument(
+                        fmt::format("the acquired index is beyond array size, field: {}, index: {}, array size: {}",
```

Comment:
In standard JSONPath, that could return a null ? instead of report error?
Anyway, we should try best to keep compatible with JSONPath?

Reply:
This has been changed to `Status::NotFound`, which is the same as json path not found.